### PR TITLE
Bug 1251988 - Add UI features to Treeherder to enable comparison of structured logs in WPT Viewer

### PR DESCRIPTION
**Pull request by martiansideofthemoon**
Attempt to fix  https://github.com/mozilla/wptview/issues/78

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1321)

### CODE REVIEW

File: ui/plugins/job_details/main.html [hunk 1]
```diff
@@ -5,6 +5,7 @@
       <span ng-if="line.content_type != 'TalosResult'" ng-switch on="line.content_type">
         <a ng-switch-when="link" title="{{line.value}}"
            href="{{line.url}}" target="_blank">{{line.value}}</a>
+        <a title="{{line.value}}" href="http://jgraham.github.io/wptview/#/?urls={{line.url}},{{line.value}}">Open</a>
```
Comment (wlach) [inline ui/plugins/job_details/main.html]: I think this is fine as a start. At some point we might want to make to define this in an angular directive so we can re-use this logic as needed.

**Comment by jgraham** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[ui/plugins/job_details/main.html, line 8 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/1321#-KBhnSN-3VZ8Xy3ixkFI-r1-8:-KBx-KQWdZSE7CHSk42m:519555377) ([raw file](https://github.com/mozilla/treeherder/blob/eb05ddc075375b563763dd8b798f14af4ded2ed0/ui/plugins/job_details/main.html#L8)):*
Isn't `ng-if` better than `ng-show` here? Also I think you want to encode the data that you are putting into the url.

I didn't carefully check the usage of `job_details`

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/1321)*

**Comment by wlach** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[ui/plugins/job_details/main.html, line 8 \[r1\]](https://reviewable.io:443/reviews/mozilla/treeherder/1321#-KBhnSN-3VZ8Xy3ixkFI-r1-8:-KBxqGfdFUfrB_FCR9_y:-716686762) ([raw file](https://github.com/mozilla/treeherder/blob/eb05ddc075375b563763dd8b798f14af4ded2ed0/ui/plugins/job_details/main.html#L8)):*
Yeah, ng-if would be very maginally better here: http://stackoverflow.com/a/21870119

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/1321)*

**Comment by martiansideofthemoon** [comment]
@wlach @jgraham , I'd need a filter for `encodeURIComponents`. In which JS file could I add it?

**Comment by wlach** [comment]
Angular filters should go in `ui/js/filter.js`

**Comment by martiansideofthemoon** [comment]
@wlach made the changes

File: ui/plugins/job_details/main.html [hunk 2]
```diff
@@ -5,6 +5,8 @@
       <span ng-if="line.content_type != 'TalosResult'" ng-switch on="line.content_type">
         <a ng-switch-when="link" title="{{line.value}}"
            href="{{line.url}}" target="_blank">{{line.value}}</a>
+
+        <a title="{{line.value}}" href="http://mozilla.github.io/wptview/#/?urls={{line.url | encodeURIComponent}},{{job_details[job_details.length - 1].value | encodeURIComponent}}" ng-if="line.value.endsWith('raw.log')"><br>Filter results in wptview</a>
```
Comment (wlach) [inline ui/plugins/job_details/main.html]: Putting it on a seperate line is indeed confusing. How about something like this:

```html
<span ng-if="line.value.endsWith('raw.log')">
- <a title="{{line.value}}" href="http://mozilla.github.io/wptview/#/?urls={{line.url | encodeURIComponent}},{{job_details[job_details.length - 1].value | encodeURIComponent}}" ng-if="line.value.endsWith('raw.log')">open in wptview</a>
</span>
```

I'm still not 100% sure the results are valid -- I can't seem to load anything from the URLs it generates:

http://mozilla.github.io/wptview/#/?urls=http:%2F%2Fmozilla-releng-blobs.s3.amazonaws.com%2Fblobs%2Fmozilla-inbound%2Fsha512%2Ff67c282527e4015e922f20eb71db5881e7d5448db052dfe05ce5e4f233bfd9803f7d5f1c832593793137a8f339cc2c950684c758186487bbf280f0b9e245e793,Rev7%20MacOSX%20Yosemite%2010.10.5%20mozilla-inbound%20debug%20test%20web-platform-tests-e10s-8

Am I missing something?

Comment (martiansideofthemoon) [inline ui/plugins/job_details/main.html]: This link successfully loaded in wptview for me. One can see the results after adding a filter

Comment (martiansideofthemoon) [inline ui/plugins/job_details/main.html]: I think this is a great fix as well! :smile:

**Comment by martiansideofthemoon** [comment]
@wlach , @jgraham there was a small problem with wptview. Fixed it in [https://github.com/mozilla/wptview/commit/302855982272281534dde60ed337b2265341f821](url)

**Comment by wlach** [comment]
Merged in 13639885f4cacbbcc398b5af32284845310d39b6

**Pull request closed without merging**